File: Python/Prob9.py
#pretty sure  this workes, but the haskell one got there first.

def isTrip(a,b,c):
	return (a**2 + b**2 == c**2)

def sumP(a,b,c):
	return (a+b+c==1000)

a=b=c= range(1,1000)

for i in a:
	for j in b:
		for k in c:
			if isTrip(i,j,k) and sumP(a,b,c):
				print a*b*c

File: Haskell/prob9.hs
--takes a while, but gets there eventually
isTrip a b c = (a^2 +b^2 == c^2) && (a<b && b<c)

sum' a b c = a+b+c==1000

t = [a*b*c | a<-[1..1000], b<-[1..1000], c<-[1..1000], isTrip a b c, sum' a b c]

main = print $ head t